### DISCUSSION TITLE
add friendly name IP address to the entity

### DISCUSSION THREAD

**Discussion by sevagjb** [vmakeev/huawei_mesh_router#44] [Ideas]
It will be very easy if you add friendly name the IP addresses to the entities then we can sort it smoothly,
i like to hear from you if that is possible

Thanks a lot its already very useful add-one

**Comment by vmakeev** [comment]
Hi!

Which entities exactly do you mean?

**Comment by sevagjb** [comment]
Greeting my friend,

in HA i get all the connected devices as entities, it naemd with the mac
addresses as entity ID  it will be very useful if you can add the ip
address to the entity also as friendly name


Thanks

**Comment by vmakeev** [comment]
I think I understand what you're talking about.

Currently, the `friendly name`, like the entity identifier, is taken from the name of the connected device in the router itself. 

If the device name is not assigned in the router, the integration tries to get the `HostName` of the device itself. And only if this attempt does not lead to anything, then it generates a name and Id according to the template `device_<MAC_ADDRESS>`

Renaming devices in the router's web interface is quite simple, you will need exactly three mouse clicks:

![image](https://user-images.githubusercontent.com/13693034/219270398-0a950447-d061-4327-b752-8801e24a0055.png)
![image](https://user-images.githubusercontent.com/13693034/219270413-be70cd04-4fd7-4490-b375-718c38271d7b.png)

-----

Result:

![image](https://user-images.githubusercontent.com/13693034/219274158-942da660-9022-4954-afc6-e8b70e0fc73c.png)


Using IP addresses for the display name of entities does not seem quite correct: IP addresses can be dynamic, and the same entity may have a different name at different times, although no renaming should have been done on your part

**Comment by sevagjb** [comment]
Ok, thanks.
i already busy in renaming my devices

**Comment by vmakeev** [comment]
Nice :)

Let me know if there are any difficulties or inconveniences when working with integration
